crates/pixi_core/src/lock_file/update.rs

```diff
@@ -1358,13 +1408,40 @@ impl<'p> UpdateContext<'p> {
                     .unwrap_or_default();
 
                 // Spawn a task to solve the group.
+                // Determine override pinned source for current package when performing
+                // a targeted update that does not include the current package.
+                let override_pinned = (|| {
```

Comment:
Can we make sure this is required and we can't let the solver take care of this by giving it this information through the locked_group_records?